### PR TITLE
Various doc improvements in cirq/ops

### PR DESCRIPTION
**Pull request by Strilanc**

### CODE REVIEW

File: cirq/ops/raw_types.py [hunk 1]
```diff
@@ -364,7 +364,14 @@ def _qid_shape_(self) -> Tuple[int, ...]:
...
-    def with_qubits(self, *new_qubits: Qid) -> 'Operation':
+    def with_qubits(self, *new_qubits: 'cirq.Qid') -> 'cirq.Operation':
+        """Returns the same operation, but applied to different qubits.
+
+        Args:
+            new_qubits: The new qubits to apply the operation to. The order must
+                exactly match the order of qubits returned from the operation's
+                `qubits` property.
+        """
         pass
```
Comment (viathor) [inline cirq/ops/raw_types.py]: pylint: remove `pass`

File: cirq/ops/pauli_interaction_gate.py [hunk 2]
```diff
@@ -37,6 +37,7 @@
 class PauliInteractionGate(eigen_gate.EigenGate,
                            gate_features.InterchangeableQubitsGate,
                            gate_features.TwoQubitGate):
+    """Can represent a CZ conjugated by arbitrary single qubit Cliffords."""
```
Comment (viathor) [inline cirq/ops/pauli_interaction_gate.py]: "Can represent a" is unnecessary.

Comment (Strilanc) [reply cirq/ops/pauli_interaction_gate.py]: Removed.

File: cirq/ops/raw_types.py (hunk 1 above)
Comment (Strilanc) [reply cirq/ops/raw_types.py]: Done.

**Pull request merged**